### ISSUE TITLE
[MJAVADOC-416] java.lang.ClassCastException: com.sun.tools.javadoc.ClassDocImpl cannot be cast to com.sun.javadoc.AnnotationTypeDoc

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#732]
**[Alex Collins](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=alex.e.c)** opened **[MJAVADOC-416](https://issues.apache.org/jira/browse/MJAVADOC-416?redirect=false)** and commented

This is a new issue in 2.10, it does not exist in 2.9.1. I'm raising it as blocker, and we cannot use this version.

It does not happen on the basic Maven archetype.

Reproduce:

```
mvn org.apache.maven.plugins:maven-javadoc-plugin:2.10:jar
```

Can you can compare this against 2.9.1 which works OK:

```
mvn org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar
```

Stack trace;

```
[ERROR] java.lang.ClassCastException: com.sun.tools.javadoc.ClassDocImpl cannot be cast to com.sun.javadoc.AnnotationTypeDoc
[ERROR] at com.sun.tools.javadoc.AnnotationDescImpl.annotationType(AnnotationDescImpl.java:46)
[ERROR] at com.sun.tools.doclets.internal.toolkit.util.Util.isDeprecated(Util.java:811)
[ERROR] at com.sun.tools.doclets.formats.html.PackageWriterImpl.writeClassesSummary(PackageWriterImpl.java:114)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.buildClassSummary(PackageSummaryBuilder.java:175)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:597)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.invokeMethod(PackageSummaryBuilder.java:81)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.AbstractBuilder.build(AbstractBuilder.java:90)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.buildSummary(PackageSummaryBuilder.java:148)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:597)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.invokeMethod(PackageSummaryBuilder.java:81)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.AbstractBuilder.build(AbstractBuilder.java:90)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.buildPackageDoc(PackageSummaryBuilder.java:106)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:597)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.invokeMethod(PackageSummaryBuilder.java:81)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.AbstractBuilder.build(AbstractBuilder.java:90)
[ERROR] at com.sun.tools.doclets.internal.toolkit.builders.PackageSummaryBuilder.build(PackageSummaryBuilder.java:92)
[ERROR] at com.sun.tools.doclets.formats.html.HtmlDoclet.generatePackageFiles(HtmlDoclet.java:183)
[ERROR] at com.sun.tools.doclets.internal.toolkit.AbstractDoclet.startGeneration(AbstractDoclet.java:120)
[ERROR] at com.sun.tools.doclets.internal.toolkit.AbstractDoclet.start(AbstractDoclet.java:64)
[ERROR] at com.sun.tools.doclets.formats.html.HtmlDoclet.start(HtmlDoclet.java:42)
[ERROR] at com.sun.tools.doclets.standard.Standard.start(Standard.java:23)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:597)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.invoke(DocletInvoker.java:269)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.start(DocletInvoker.java:143)
[ERROR] at com.sun.tools.javadoc.Start.parseAndExecute(Start.java:340)
[ERROR] at com.sun.tools.javadoc.Start.begin(Start.java:128)
[ERROR] at com.sun.tools.javadoc.Main.execute(Main.java:41)
[ERROR] at com.sun.tools.javadoc.Main.main(Main.java:31)
```

---

**Affects:** 2.10

**Issue Links:**
- [MJAVADOC-407](https://issues.apache.org/jira/browse/MJAVADOC-407) Cannot parse annotations when generating javadoc
 (_**"duplicates"**_)

**Comment by jira-importer** [comment]
**[Alex Collins](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=alex.e.c)** commented

Suspect change:

```
+++ b/maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/AbstractJavadocMojo.java
@@ -3531,7 +3531,8 @@ public abstract class AbstractJavadocMojo
             javadocExe =
                 new File( SystemUtils.getJavaHome() + File.separator + ".." + File.separator + "sh", javadocCommand );
         }
-        else if ( SystemUtils.IS_OS_MAC_OSX )
+        // For Apple's JDK 1.6.x (and older?) on Mac OSX
+        else if ( SystemUtils.IS_OS_MAC_OSX && SystemUtils.JAVA_VERSION_FLOAT < 1.7f )
         {
             javadocExe = new File( SystemUtils.getJavaHome() + File.separator + "bin", javadocCommand );
         }

```

**Comment by jira-importer** [comment]
**[Alex Collins](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=alex.e.c)** commented

This appears to be fixed in 2.10.1. We're locking to 2.9.x now.
